java-checks/src/test/files/checks/naming/KeywordAsIdentifierCheck_enum.java

```diff
@@ -0,0 +1,21 @@
+class A {
+  int foo = 0;
+  int enum = 0; // Noncompliant [[sc=7;ec=11]] {{Use a different name than "enum".}}
+
+  enum mynum {RED, GREEN, YELLOW};
```

Comment:
Unfortunately this is still not valid Java code of any version:
* not Java 5 or greater, because `enum` is used as identifier at line 3
* not Java 4 or lower, because of attempt to declare `enum` at lines 5 and 19

Reply:
Good point. Sorry for missing this.